internal/dcache/rpc/server/handler.go

```diff
@@ -47,43 +57,894 @@ var _ service.ChunkService = &ChunkServiceHandler{}
 // ChunkServiceHandler struct implements the ChunkService interface
 type ChunkServiceHandler struct {
 	locks *common.LockMap
-	// more fields can be added here as needed
+
+	// map to store the rvID to rvInfo mapping
+	// rvIDMap contains any and all cluster awareness information that a node needs to have,
+	// things like what all RVs are hosted by this node, state of each of those RVs,
+	// what all MVs are hosted by these RVs, state of those MVs, etc.
+	// It is initially created from the clustermap which is the source of truth regarding cluster information,
+	// and once the cluster is working it's updated using various RPCs.
+	// Note that any time Cluster Manager needs to update clustermap, before publishing the updated clustermap,
+	// it'll send our one or more RPCs to update the rvIDMap info in all the affected nodes,
+	// thus rvIDMap always contains the latest info and hence is used by RVs to fail requests
+	// which might be sent by nodes having a stale clustermap.
+	rvIDMap map[string]*rvInfo
+}
+
+type rvInfo struct {
+	rvID     string       // id for this RV
+	rvName   string       // rv0, rv1, etc.
+	cacheDir string       // cache dir path for this RV
+	mvMap    sync.Map     // sync map of MV id against mvInfo
+	mvCount  atomic.Int64 // count of MVs for this RV, this should be updated whenever a MV is added or removed from the sync map
+}
+
+type mvInfo struct {
+	mu           sync.Mutex
+	mvName       string   // mv0, mv1, etc.
+	componentRVs []string // sorted list of component RVs for this MV
+
+	// count of chunk operations (get, put or remove) for this MV.
+	// This is used to block the end sync call till all the ongoing chunk operations are completed.
+	chunkOpsCnt atomic.Int64
+
+	// flag to block chunk operations (get, put or remove) for this MV.
+	// This flag is enabled in the end sync call to pause new chunk operations in the MV.
+	// When the end sync call is completed, this flag is disabled.
+	blockOpsFlag atomic.Bool
+
+	syncInfo // sync info for this MV
+}
+
+type syncInfo struct {
```

Comment:
we should also store reservedSpace inside the syncInfo struct. This must be reduced from available space of the RV to get true available space. This means while sync is going on we might be overestimating the space utilization, but I think that's ok and later we can make it more accurate.

Reply:
Added TODO